benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015, The WGS-HGT Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+#
+# This script allows to simulate horizontal gene transfer by combining genes
+# from raw nucleotide genomes.
+#
+# Input: the inputs are two genomes, donor and recipient (in GenBank format)
+#        and options regarding the number of HGTs to simulate and the types of
+#        HGTs allowed (orthologous replacement and novel gene acquisition
+#        supported)
+#
+# Algorithm: all protein coding sequences (cds) for both genomes are extracted
+#            using scikit-bio. For orthologous gene replacement, orthologous
+#            genes are determined using OrthoFinder. A python function
+#            combines genes from the two genomes (orthologous and novel).
+#
+# Output:    the resulting genomes in raw nucleotide and protein coding
+#            sequences FASTA formats and a log file showing the nucleotide
+#            positions of spiked genes
+#
+# Dependencies: this script requires OrthoFinder
+#
+# Additional information:
+#   1. Emms, D. and Kelly, S. (2015). OrthoFinder: solving fundamental biases
+#      in whole genome comparisons dramatically improves orthogroup inference
+#      accuracy, GenomeBiology, 16:157
+#
+
+import sys
+import click
+from os import makedirs
+from os.path import exists, basename, join, splitext
+import subprocess
+import time
+import glob
+import random
+from operator import itemgetter
+
+from skbio import Sequence
+
+
+def extract_genbank(genbank_fp, verbose=False):
+    """Extract protein coding sequences from GenBank record.
+    Parameters
+    ----------
+    genbank_fp: string
+        file path to genome in GenBank format
+
+    Returns
+    -------
+    seq: skbio.sequence.Sequence
+        Sequence object
+    genes: dictionary
+        a dictionary of genes (CDS) and their info, with the key being the
+        protein IDs and the value being a 4-element list including the
+        translated sequence, the start and end positions in the genome
+    """
+    genes = {}
+    if verbose:
+        sys.stdout.write("\tParse GenBank record ...\n")
+    seq = Sequence.read(genbank_fp, format='genbank')
+    if verbose:
+        sys.stdout.write("\t\tDone.\n")
+    for feature in list(seq.interval_metadata.features.keys()):
+        if feature['type_'] == 'CDS':
+            protein_id = feature['protein_id']
+            translation = feature['translation']
+            strand = '+'
+            if feature['rc_']:
+                strand = '-'
+            loc = seq.interval_metadata.features[feature]
+            start_pos = loc[0][0]
```

Comment:
could probably do the expansion inline, but i can't remember if pep8 hates this:

```python
start_pos, end_pos = loc[0]
```